cmd/tendermint/commands/init.go

```diff
@@ -99,5 +106,87 @@ func initFilesWithConfig(config *cfg.Config) error {
 		logger.Info("Generated genesis file", "path", genFile)
 	}
 
+	if err := InitIpfs(config); err != nil {
+		return err
+	}
+
 	return nil
 }
+
+// TODO move into separate file
+func InitIpfs(config *cfg.Config) error { // add counter part in ResetAllCmd
+	// init IPFS config with params from config.IPFS
+	// and store in config.IPFS.ConfigRootPath
+	repoRoot := config.IPFSRepoRoot()
+	if !fsrepo.IsInitialized(repoRoot) {
+		var conf *ipfscfg.Config
+
+		identity, err := ipfscfg.CreateIdentity(os.Stdout, []options.KeyGenerateOption{
+			options.Key.Type(options.Ed25519Key),
+		})
+		if err != nil {
+			return err
+		}
+
+		logger.Info("initializing IPFS node", "ipfs-path", repoRoot)
+
+		if err := tmos.EnsureDir(repoRoot, 0700); err != nil {
+			return err
+		}
+
+		conf, err = ipfscfg.InitWithIdentity(identity)
+		if err != nil {
+			return err
+		}
+
+		applyFromTmConfig(conf, config.IPFS)
+		if err := setupPlugins(repoRoot); err != nil {
+			return err
+		}
+
+		if err := fsrepo.Init(repoRoot, conf); err != nil {
+			return err
+		}
+	} else {
+		logger.Info("IPFS was already initialized", "ipfs-path", repoRoot)
+	}
+	return nil
+}
```

Comment:
Here's a slight alternative that removes one layer of nesting and the else statement
```suggestion
func InitIpfs(config *cfg.Config) error { // add counter part in ResetAllCmd
	// init IPFS config with params from config.IPFS
	// and store in config.IPFS.ConfigRootPath
	repoRoot := config.IPFSRepoRoot()
	if fsrepo.IsInitialized(repoRoot) {
		logger.Info("IPFS was already initialized", "ipfs-path", repoRoot)
	}

	var conf *ipfscfg.Config

	identity, err := ipfscfg.CreateIdentity(os.Stdout, []options.KeyGenerateOption{
		options.Key.Type(options.Ed25519Key),
	})
	if err != nil {
		return err
	}

	logger.Info("initializing IPFS node", "ipfs-path", repoRoot)

	if err := tmos.EnsureDir(repoRoot, 0700); err != nil {
		return err
	}

	conf, err = ipfscfg.InitWithIdentity(identity)
	if err != nil {
		return err
	}

	applyFromTmConfig(conf, config.IPFS)
	if err := setupPlugins(repoRoot); err != nil {
		return err
	}

	if err := fsrepo.Init(repoRoot, conf); err != nil {
		return err
	}

	return nil
}
```

Reply:
>  I swear, I have nothing against else statements, this is just a tad more readable IMHO.

Then, let them else-statements live :-D 

No, seriously, I agree. Removing that extra nesting makes the code more readable 👍🏼